packages/organizations/origin-organization-irec-api/src/beneficiary/handlers/add-organization-beneficiary.handler.ts

```diff
@@ -0,0 +1,47 @@
+import { Repository } from 'typeorm';
+import { CommandBus, CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { InjectRepository } from '@nestjs/typeorm';
+import { UserService } from '@energyweb/origin-backend';
+
+import { AddOrganizationBeneficiaryCommand, GetBeneficiaryCommand } from '../commands';
+import { Beneficiary } from '../beneficiary.entity';
+import { NotFoundException } from '@nestjs/common';
+import { BeneficiaryDTO } from '../dto/beneficiary.dto';
+
+@CommandHandler(AddOrganizationBeneficiaryCommand)
+export class AddOrganizationBeneficiaryHandler
+    implements ICommandHandler<AddOrganizationBeneficiaryCommand>
+{
+    constructor(
+        @InjectRepository(Beneficiary)
+        private readonly repository: Repository<Beneficiary>,
+        private readonly userService: UserService,
+        private readonly commandBus: CommandBus
+    ) {}
+
+    public async execute({
+        ownerOrganizationId,
+        irecBeneficiaryId
+    }: AddOrganizationBeneficiaryCommand): Promise<BeneficiaryDTO> {
+        const platformAdmin = await this.userService.getPlatformAdmin();
```

Comment:
If `ownerOrganizationId` is meant to be the id of the organization that the user belongs to then getPlatformAdmin is the wrong method as it returns the organization that owns the platform, not the users' organization.